br/pkg/lightning/backend/external/testutil.go

```diff
@@ -127,3 +127,37 @@ func splitDataAndStatFiles(datas []string, stats []string) ([][]string, [][]stri
 	}
 	return dataGroup, statGroup
 }
+
+// split data&stat files, startKeys and endKeys into groups for new merge step.
+func splitDataStatAndKeys(datas []string, stats []string, multiStats []MultipleFilesStat) ([][]string, [][]string, []dbkv.Key, []dbkv.Key) {
+	startKeys := make([]dbkv.Key, 0, 10)
+	endKeys := make([]dbkv.Key, 0, 10)
+	i := 0
+	for ; i < len(multiStats)-1; i += 2 {
+		startKey := BytesMin(multiStats[i].MinKey, multiStats[i+1].MinKey)
+		endKey := BytesMax(multiStats[i].MaxKey, multiStats[i+1].MaxKey)
+		endKey = dbkv.Key(endKey).Next().Clone()
+		startKeys = append(startKeys, startKey)
```

Comment:
we're calculating start/end keys by sum up every 2 multiStats, but calculate dataGroups by every 1000 files, seems they don't relate to each other?

Reply:
multistats have 500 files, so I calculate 1000 files.